### PR TITLE
Added --public flag for publish

### PR DESCRIPTION
**Pull request by kennethlimcp**
- defaults to private for publishing
- Fixes: https://github.com/spark/particle-cli/issues/132
- This should not be merged in until i write up some docs and confirm the syntax with @nexxy

Signed-off-by: Kenneth Lim <kennethlimcp@gmail.com>

### CODE REVIEW

**Comment by kennethlimcp** [comment]
```

KENMBP:particle-cli kennethlimcp$ node app.js publish my-event
Published private event: my-event

KENMBP:particle-cli kennethlimcp$ node app.js publish my-event --public
Published public event: my-event

KENMBP:particle-cli kennethlimcp$ node app.js publish my-event someData --public 
Published public event: my-event

KENMBP:particle-cli kennethlimcp$ node app.js publish my-event someData
Published private event: my-event

KENMBP:particle-cli kennethlimcp$ 
```

**Comment by m-mcgowan** [comment]
I'd prefer for events to be public by default (like the firmware API) and add a `--private` flag. (Keep `--public` in case people want to be explicit about it to make scripts easier to read.)

**Comment by kennethlimcp** [comment]
@m-mcgowan, i changed it to the `--private` flag that sets it to private and defaults to `private: true` but skipped keeping the `--public` flag everything without `--private` flag will be just set to public. 

Thoughts?

**Comment by m-mcgowan** [comment]
Sounds good, but for me I'm a fan of having an explicit option for every default option. But that's just a matter of taste, so you're free to leave it.

File: commands/PublishCommand.js [hunk 1]
```diff
@@ -66,7 +67,17 @@ PublishCommand.prototype = extend(BaseCommand.prototype, {
 			return -1;
 		}
 
-		api.publishEvent(eventName, data);
+		if(data == "--private" && setPrivate === undefined){
+			setPrivate = true;
+			data = "";
```
Comment (brycekahle) [inline commands/PublishCommand.js]: This should be `data = undefined;` to maintain backwards compatibility.

**Comment by brycekahle** [comment]
@kennethlimcp Thanks for the PR! Please fix the one issue and then I will merge.

**Pull request merged**